lib/Onyx.js

```diff
@@ -874,8 +891,14 @@ function set(key, value) {
         return remove(key);
     }
 
+    if (hasPendingMergeForKey(key)) {
+        Logger.logAlert(`Onyx.set() called after Onyx.merge() for key: ${key}. It is recommended to use set() or merge() not both.`);
+    }
+
     // This approach prioritizes fast UI changes without waiting for data to be stored in device storage.
-    broadcastUpdate(key, value, 'set');
+    const hasChanged = broadcastUpdate(key, value, 'set');
+
+    if (!hasChanged) { return Promise.resolve(); }
```

Comment:
```suggestion
    if (!hasChanged) { 
        return Promise.resolve();
     }
```
Can you also please add a comment about _why_ this early return should be here?

Reply:
+1 to the comment

Reply:
Added comments! We returned a resolved promise before too, when the value hasn't changed. That's why i added this back, same for `merge`